pkg/reboot/rebooter.go

```diff
@@ -2,14 +2,16 @@ package reboot
 
 import (
 	"os/exec"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"time"
 
 	"github.com/go-logr/logr"
 	"github.com/medik8s/self-node-remediation/pkg/watchdog"
 )
 
 type Rebooter interface {
 	// Reboot triggers a node reboot
-	Reboot() error
+	Reboot() (ctrl.Result, error)
```

Comment:
I don't think we need the `Result` here, and the logic for setting `Requeue` if needed.
The reconciler is already calling `Reboot()` until it detected the reboot. I think just the `Requeue` over there needs to be changed to a shorter interval than now.

Reply:
>I think just the Requeue over there needs to be changed to a shorter interval than now.

I think I'm missing something, I assume you refer to requeue [here](https://github.com/medik8s/self-node-remediation/blob/0eb8d6ca7aa1e65371bcc4e6f89fde3758341455/controllers/selfnoderemediation_controller.go#L418) instead of in the reboot method (at the moment there is no timeout). 

>The reconciler is already calling Reboot() until it detected the reboot

This is true, however our use-case is different, we need to allow the watchdog some time to reboot before applying a safeguard software reboot, at the moment the fields  which are needed (i.e mainly watchdog inner params) for this logical decision (i.e applying or not the software safeguard reboot)  are not exposed at the higher scope.

The code can be refactored in a such a way that those fields will be available in a higher level scope, but IMO it'll be wrong.

Reply:
Indeed, Done :+1: